src/plugins/smith_waterman/src/SmithWatermanAlgorithm.cpp

```diff
@@ -173,7 +173,7 @@ void SmithWatermanAlgorithm::calculateMatrixForMultipleAlignmentResult() {
 
     n = pat_n * 2;
     int dirn = (4 + pat_n + 3) >> 2;
-    int *buf, *matrix = (int *)malloc(n * sizeof(int) + pat_n * 0x80 + matrixLength * dirn);
+    int *buf, *matrix = new int[n * sizeof(int) + pat_n * 0x80 + matrixLength * dirn];
```

Comment:
Why can't you keep the old code and just add a check: if (matrix === nullptr) throw bad-alloc... ?

+ there is another place like this with malloc() in the file.

Reply:
Ok. Fix with minimal changes